packages/toolpad-app/src/toolpad/DomLoader.tsx

```diff
@@ -50,6 +60,15 @@ export type DomAction =
       type: 'DOM_UPDATE';
       updater: (dom: appDom.AppDom) => appDom.AppDom;
       selectedNodeId?: NodeId | null;
+      view?: DomView;
```

Comment:
Why can't we make switching a view its own action? It will push a single entry on the stack, so...?

Reply:
I created an action just for switching a view, but for some actions, such as creating a new page or connection, the UX feels nicer if the switch and the creation happen in the same action. Other than a few cases like that they work well as separate actions.

Reply:
Right now I'm using the `currentView` in the DOM state as the source of truth for navigation routes such as connection or code components, and for the state in some modals such as the query editor...
We could make it work the other way around, and use the route or state as the source of truth for the current view, I've thought about that, I guess it should work...

Reply:
We need simultaneous DOM and view updates sometimes because for example when deleting a page: "delete page" and then "set default page as current view", when reversed on undo, would first set the deleted page as the current view (without it existing), and only then recreate it. I changed the way the view can be passed to the `.update` function to try to make it more readable.